src/main/java/com/google/opensesame/projects/ProjectEntity.java

```diff
@@ -0,0 +1,116 @@
+package com.google.opensesame.projects;
+
+import com.google.appengine.api.datastore.PropertyContainer;
+import com.google.opensesame.github.GitHubGetter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.kohsuke.github.GHRepository;
+import org.kohsuke.github.GitHub;
+
+/** The class used to interact with the project entities in data storage. */
+public class ProjectEntity {
+  public static final String REPOSITORY_ID_KEY = "repositoryId";
+  public static final String NUM_MENTORS_KEY = "numMentors";
+  public static final String NUM_INTERESTED_USERS_KEY = "numInterestedUsers";
+  public static final String MENTOR_IDS_KEY = "mentorIds";
+  public static final String INTERESTED_USER_IDS_KEY = "interestedUserIds";
+
+  /**
+   * Creates a new ProjectEntity from a GitHub repository name.
+   *
+   * @param repositoryName
+   * @return Returns the new ProjectEntity or null if the repository name is invalid.
+   * @throws IOException
+   */
+  public static ProjectEntity fromRepositoryName(String repositoryName) throws IOException {
+    GitHub gitHub = GitHubGetter.getGitHub();
+    GHRepository repository;
+    try {
+      repository = gitHub.getRepository(repositoryName);
+    } catch (IOException e) {
+      return null;
+    }
+
+    return new ProjectEntity(
+        String.valueOf(repository.getId()), new ArrayList<String>(), new ArrayList<String>());
+  }
+
+  /**
+   * Creates a ProjectEntity from the properties of an existing entity in datastore. Assumes that
+   * that all of the required keys are populated in the properties.
+   *
+   * @param properties The properties of an existing datastore entity.
+   * @return Returns the new ProjectEntity.
+   */
+  public static ProjectEntity fromProperties(Map<String, Object> properties) {
+    // Warning is suppressed because it is a precondition that the properties are from a
+    // project entity.
+    @SuppressWarnings({"unchecked"})
+    ProjectEntity newEntity =
+        new ProjectEntity(
+            (String) properties.get(REPOSITORY_ID_KEY),
+            (List<String>) properties.get(MENTOR_IDS_KEY),
+            (List<String>) properties.get(INTERESTED_USER_IDS_KEY));
+    return newEntity;
+  }
+
+  public final String repositoryId;
+  public List<String> mentorIds;
+  public List<String> interestedUserIds;
+
+  /**
+   * Creates a new ProjectEntity object.
+   *
+   * @param repositoryId
+   * @param mentorIds
+   * @param interestedUserIds
+   */
+  public ProjectEntity(
+      String repositoryId, List<String> mentorIds, List<String> interestedUserIds) {
+    this.repositoryId = repositoryId;
+    this.mentorIds = mentorIds;
+    this.interestedUserIds = interestedUserIds;
+  }
+
```

Comment:
Maybe an ```addMentor()``` function if that makes the flow easier.

Reply:
That's a good point. Both the `mentorIds` and `interestedUserIds` lists are publicly accessible at the moment, but I'm wondering if it'd be more safe to make them both private and only allow for adding and removing. Hmm... What do you think? Also Bill and Joe feel free to chime in here on what you would consider the best practices.

Reply:
Yeah having an addMentor function might be convenient, maybe not so much here but related to the Datastore? But it's also probably fine just to use the public field and save the whole project entity again for now.

Reply:
@billyeh You make a good point, however Datastore is designed such that updating an entity [involves saving it back to the Datastore](https://cloud.google.com/appengine/docs/standard/java/datastore/creating-entities#Updating_entities) which is why I thought it was best to create this class as a clean way to interface with the properties of an existing Datastore entity.

Reply:
Sounds good! nit: might just want to name `addProjectEntity` `putProjectEntity` since it adds or updates.